lunatrace/bsl/backend/src/graphql-yoga/resolvers/create-pull-request-for-vulnerability.ts

```diff
@@ -0,0 +1,142 @@
+
+    const vulnerabilityId = args.vulnerability_id;
+    if (!vulnerabilityId) {
+      throw new GraphQLYogaError('No vulnerability id provided');
+    }
+
+    const projectId = args.project_id;
+    if (!projectId) {
+      throw new GraphQLYogaError('No project id provided');
+    }
+
+    const oldPackageSlug = args.old_package_slug;
+    if (!oldPackageSlug) {
+      throw new GraphQLYogaError('No old package slug provided');
+    }
+
+    const newPackageSlug = args.new_package_slug;
+    if (!newPackageSlug) {
+      throw new GraphQLYogaError('No new package slug provided');
+    }
+
+    const packageManifestPath = args.package_manifest_path;
+    if (!packageManifestPath) {
+      throw new GraphQLYogaError('No package manifest path provided');
+    }
+
+    // Strips out the actual package.json/yarn.lock/package-lock.json from the path since we just need the folder.
+    const normalizedPackageManifestPath = packageManifestPath
+      .replace(/package\.json$/, '')
+      .replace(/yarn\.lock$/, '')
+      .replace(/package-lock\.json$/, '');
+
+    await checkProjectIsAuthorizedOrThrow(projectId, ctx);
+
+    const installationIdResult = await hasura.GetGitHubInstallationIdFromProjectId({
+      project_id: projectId,
+    });
+
+    if (installationIdResult.projects.length === 0) {
+      throw new GraphQLYogaError('No project found when fetching installation id');
+    }
+
+    const installationId = installationIdResult.projects[0].organization?.installation_id;
+
+    if (!installationId) {
+      throw new GraphQLYogaError('No installation id found for project');
+    }
+
+    const githubRepo = installationIdResult.projects[0].github_repository?.git_url;
+
+    if (!githubRepo) {
+      throw new GraphQLYogaError('No github repo found for project');
+    }
+
+    const { owner, repo } = splitGitHubRepoPath(githubRepo);
+
+    const installationAccessTokenRes = await getInstallationAccessToken(installationId);
+    if (installationAccessTokenRes.error) {
+      log.error('Failed to fetch installation access token', { installationAccessTokenRes });
+      throw new Error('Error fetching installation access token');
+    }
+
+    const installationAccessToken = installationAccessTokenRes.res;
+
+    const octokit = new PullRequestOctokit({
+      auth: installationAccessToken,
+    });
+
+    const pullRequestResult = await replacePackageAndFileGitHubPullRequest(
+      octokit,
+      owner,
+      repo,
+      normalizedPackageManifestPath,
+      // TODO: Make this less hacky and brittle
+      packageManifestPath.endsWith('package-lock.json') ? 'npm' : 'yarn',
+      oldPackageSlug,
+      newPackageSlug
+    );
+
+    // TODO: Actually implement this endpoint instead of this stub
+    return {
+      success: true,
+      pullRequestUrl: pullRequestResult.pullRequestUrl,
+    };
+  } catch (error) {
+    // TODO: temporary error handler until i figure out how to deal with global errors in yoga which seems maybe impossible
```

Comment:
It's impossible unless we drop yoga for apollo, or at least, difficult, i tried.  we need to  go to apollo anyway because yoga is just a sucky wrapper around apollo. 

Apollo stinks but it at least has that part working.

You may as well wrap this unknown error in a nice graphqlError because yoga will catch it anyway, and return something nebulous to the client. So you might as well just throw `new "GraphQLYogaError("An Unknown error occurred while attempting to update dependencies")` I have it set up so those will render out to the user in an alert.

That way it at least looks better on the client